Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Build the binary
+FROM golang:1.15 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+#COPY main.go main.go
+COPY pkg/ pkg/
+
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o k8ssandra-client pkg/main.go
```

Comment:
If we wind up with multiple hooks/jobs, do you want to use the same binary for all of them? If so, doesn't that mean we will have to introduce logic to decide what code should run based on what the hook or job is?

Reply:
Same binary, yes. The flags tell it which code path to use, in this case `-clean`. I also have another in works, `-upgrade` which updates the CRDs. The deployment files are different, but the container is the same.

Reply:
I am sure that there are pros/cons to same vs separate binary. I haven't thought about it a whole lot. What made you decide a single binary?

Reply:
Less maintenance on the docker images (single image is easier to remember to update) and also the fact that most of them use the same code or are very similar (in both parameters as well as what they do, usually through the runtime-controller k8sclient). Not to mention that client could potentially be extracted as a tool later if required.

Reply:
👍 

Thanks for providing the explanation!